sycl/include/sycl/ext/intel/esimd/detail/math_intrin.hpp

```diff
@@ -26,6 +26,49 @@
 #define __ESIMD_cpp_vec_t(T, SZ)                                               \
   __ESIMD_DNS::vector_type_t<__ESIMD_DNS::__cpp_t<T>, SZ>
 
+template <typename T> extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_exp2(T);
```

Comment:
All these declarations should be in CL/__spirv/spirv_ops.hpp
Instead of __ESIMD_raw_vec_t or vector_type_t they can use `__ocl_vec_t<T, N>`.

Reply:
I think we saw they can cause unintended use by other targets if we put it there. We could try guarding it with macros to make sure it's only hit for non-hip non-cuda though, but I'm not sure if there's a big benefit.

Reply:
Hmm, that is unexpected. Ok, then let's keep the declarations here, but also put some comment here explaining why the declarations added here and not in CL/__spirv/spirv_ops.hpp